## Maestria en Ing. de Sistemas UNAM- Procesos estocásticos . Dra: Patricia Aguilar
## Alumno: Sergio Ibarra R (414025796)
## Problemas sobre la aplicación de fundamentos de estadística.  Febrero 2023

### Problema 1

Considere los eventos excluyentes A1 y A2 cuyas probabilidades son, respectivamente, 0.40 y 0.60. Se sabe también que si B es otro evento, P(B|A1) = 0.20 y P(B|A2) = 0.05. Calcule
P(B), P(A1|B) y P(A2|B).

De la ecuación de probabilidad condicional tenemos: 
$$p(A|B)= \frac{p(A\cap B)}{p(B)}$$   

ó expresado de otra forma: 
    $$ p(A\cap B) = p(A|B)*p(B)$$

Para nuestro caso tenemos:  
    
$p(B|A1)= \frac{p(B\cap A1)}{p(A1)}$ .... (ec 1)
    

Por lo tanto, tenemos que: $p(B\cap A1) = p(B|A1)*p(A1)$ ..(ec 1.1). 


Entonces:  $p(B\cap A1) = 0.2*0.4=0.08$

##### Definamos los datos que tenemos 

In [96]:
pA1 = 0.4
pA2 = 0.6
pB_dadoA1 = 0.2
pB_dadoA2 = 0.05


##### Calculemos las probabilidades de las intersecciones de los elementos Ai (exhaustivos en unión al espacio muestral) con el elemento B

In [97]:
pB_intA1 = pB_dadoA1*pA1
pB_intA1

0.08000000000000002

Y por lo tanto la $p(B\cap A2) = p(B|A2)*p(A2)=$

In [98]:
pB_intA2 = pB_dadoA2*pA2
pB_intA2

0.03

##### Calculemos la probabilidad de B usando la "Fórmula de la probabilidad Total"

Ahora, utilizando la definición de la "Fórmula de la probabilidad Total" que indica: Sean A1 y A2 una partición colectivamente exhaustiva del espacio muestral S y sea B otro evento, entonces:

$$p(B)= p(B\cap A1)\cup p(B\cap A2)$$ 

En nuestro caso:

In [100]:
pB =  pB_intA1 + pB_intA2
pB

0.11000000000000001

##### Calculemos de nuevo las probabilidades condicionales, pero asumuendo que B es quien es el "evento seguro"

Entonces para calcular $p(A1|B)$, tenemos: 

$$p(A1|B)= \frac{p(A1\cap B)}{p(B)}$$

In [101]:
pA1_dadoB = pB_intA1/pB
pA1_dadoB

0.7272727272727273

y tambien tenemos que $p(A2|B)$ es:

$$p(A2|B)= \frac{p(A2\cap B)}{p(B)}$$

In [102]:
pA2_dadoB = pB_intA2/pB
pA2_dadoB

0.2727272727272727

### Problema 2

Cooper Realty es una pequeña compañía de bienes raíces ubicada en Albany, Nueva York, que
se especializa principalmente en listados residenciales. Recientemente se interesó en determinar
la probabilidad de que uno de sus listados se vendiera en cierto número de días. Un análisis de
las ventas de la empresa, de 800 casas en años anteriores produjo los siguientes datos:

Tabla 2.1 Precio oferta inicial vs Dias en listado hasta la venta de la empresa Cooper Realty

| Precio oferta inicial/Dias en listado hasta la venta| Menos de 30  | 31-90  |Más de 90 |TOTAL |
| :- | -: | :-: |-: |-: 
| Menos de 150,000 | 50 | 40| 10| 100
| De 150,000 a 199,000  | 20 | 150| 80| 250
| De 200,000 a $250,000 | 20 |280|100| 400
| Más de 250,000 | 10 | 30| 10|50
|TOTAL | 100 | 500| 200|800

Si A denota al evento de que una casa aparezca en el listado por más de 90 días antes de ser vendida, y B es el evento de que el precio de oferta inicial sea menor de $150,000. 

a) Calcule la probabilidad de A ∩ B

b) ¿Cuál es la probabilidad de que Cooper Realty tarde más de 90 días en vender una casa dela que acaba de firmar contrato con un precio de oferta inicial superior a $150,000?

c) Decida si son independientes o no, los eventos A y B. Justifique su respuesta.

##### Definamos  y calculemos la probabilidad de nuestros "eventos inicalmente conocidos"  
A: Probabilidad de que una casa aparezca en el listado por más de 90 días antes de ser vendida <br>
    

$$p(A)= \frac{No.casas-en-listado> 90dias}{No. total de casas en listado} = \frac{200}{800} = 0.25$$

In [103]:
pA = 200/800
pA

0.25

In [104]:
print(f"La probabilidad de A es: {pA}")

La probabilidad de A es: 0.25


B:  Probabilidad de que el precio de oferta inicial sea menor de 150,000

$$ p(B)= \frac{No.casas-en-listado< 150,000}{No. total de casas en listado} = \frac{100}{800} = 0.125 $$


In [105]:
pB = 100/800
pB

0.125

In [106]:
print(f"La probabilidad de B es: {pB}")

La probabilidad de B es: 0.125


#### a) Calcule la probabilidad de A ∩ B

A ∩ B son todas aquellas casas que aparecieron en el listado por más de 90 días y al mismo tiempo su oferta inicial era menos de 150,000 <br>
Esto se puede ver que se encuentra en la tabla (2.1) en el "elemento" [2,4] = 10 <br>
Entonces la probabilidad de ese evento es:

In [107]:
pA_intB = 10/800
pA_intB

0.0125

In [108]:
print(f"La probabilidad de A ∩ B es: {pA_intB}")

La probabilidad de A ∩ B es: 0.0125


#### b) ¿Cuál es la probabilidad de que Cooper Realty tarde más de 90 días en vender una casa de la que acaba de firmar contrato con un precio de oferta inicial superior a $150,000?

En este caso nos está preguntando por la p(A|B) que se calcularía como: <br>
     $$p(A|B)= \frac{p(A\cap B)}{p(B)}$$   
    

Lo que sería  $p(A|B)= \frac{p(A\cap B)}{p(B)}$ =  0.0125 / 0.125

In [109]:
pA_dadoB = pA_intB /pB
pA_dadoB

0.1

In [110]:
print(f"La probabilidad de tardar más de 90 días en vender una casa de la que acaba de firmar contrato con un precio de oferta inicial superior a 150,000 : {pA_dadoB}")

La probabilidad de tardar más de 90 días en vender una casa de la que acaba de firmar contrato con un precio de oferta inicial superior a 150,000 : 0.1


In [111]:
print(f"Es decir, la probabilidad de p(A|B) es: {pA_dadoB}")

Es decir, la probabilidad de p(A|B) es: 0.1


#### c) Decida si son independientes o no, los eventos A y B. Justifique su respuesta.

Debido a que: 
    $$p(A)\neq p(A|B)$$


Entonces A y B NO son eventos independientes, pues la ocurrencia de uno SI afecta la probabilidad de ocurrencia del otro

### Problema 3

En una encuesta aplicada por The Huffington Post en noviembre de 2011 a 1400 personas, con
el fin de conocer los hábitos de uso de las redes sociales y otros sitios web para compartir sus
opiniones sobre los programas de televisión, se preguntó a cada persona “si usaba o no las redes
sociales y otros sitios web para compartir sus opiniones sobre dichos programas”, se obtuvieron
los resultados que se detallan a continuación:

|.| SI usa redes sociales y otros sitios para opinar sobre programas de TV| NO usa redes sociales y otros sitios para opinar sobre programas de TV
| :- | -: | :-:  
| Mujeres | 404 | 300
| Hombres  | 332 | 364

Y que calcolando los totales tenemos: 

| .| SI usa redes sociales y otros sitios para opinar sobre programas de TV  | NO usa redes sociales y otros sitios para opinar sobre programas de TV |TOTAL |
| :- | -: | :-: |-: 
| Mujeres | 404 | 300| 704
| Hombres  | 332 | 364| 696
|TOTAL | 736 | 664| 1400

a) Calcule la probabilidad de que, al seleccionar al azar una de las respuestas recibidas corresponda a una persona que no usa redes sociales ni otros sitios web para compartir su opinión sobre programas de televisión.<br>

b) Si la respuesta a la encuesta proviene de una mujer, ¿cuál es la probabilidad de que use redes sociales y otros sitios web para compartir su opinión sobre programas de televisión?<br>

c) ¿Son independientes el género de la persona y el uso de las redes sociales y otros sitios web,con los fines mencionados en este problema? Justifique su respuesta.<br>

d) Su respuesta al inciso anterior ¿tiene alguna implicación acerca de la independencia o dependencia entre el género de la persona y el hecho de que no use las redes sociales ni otros sitios web, con los fines mencionados en este problema? Justifique su respuesta.<br>

#### a) Calcule la probabilidad de que, al seleccionar al azar una de las respuestas recibidas corresponda a una persona que no usa redes sociales ni otros sitios web para compartir su opinión sobre programas de televisión

$$p(No usar redes para opinar de TV)= \frac{No de personas que no usan redes para opinar de TV}{No. total de personas entrevistadas} = \frac{664}{1400} = 0.47$$


Definamos:<br>
-p(NR): Probabilidad de que el encuestado NO use redes para opinar de TV

In [112]:
p_NR= 664/1400
p_NR

0.4742857142857143

In [120]:
print(f" La probabilidad de seleccionar al azar una de las respuestas recibidas corresponda a una persona que no usa redes sociales ni otros sitios web para compartir su opinión sobre programas de televisión: {p_NR}")

 La probabilidad de seleccionar al azar una de las respuestas recibidas corresponda a una persona que no usa redes sociales ni otros sitios web para compartir su opinión sobre programas de televisión: 0.4742857142857143


In [121]:
print(f"Es decir, la probabilidad de p(NR) es: {p_NR}")

Es decir, la probabilidad de p(NR) es: 0.4742857142857143


#### b) Si la respuesta a la encuesta proviene de una mujer, ¿cuál es la probabilidad de que use redes sociales y otros sitios web para compartir su opinión sobre programas de televisión

Definamos:<br>
-p(M):  Probabilidad de que el encuestado sea mujer <br>
-p(UR):  Probabilidad de que el encuestado SI use redes para opinar de TV
-p(M_intUR):  Probabilidad de que el encuestado sea mujer y además SI use redes para opinar de TV

In [122]:
pM = 704/1400
pM 

0.5028571428571429

In [123]:
pUR = 736/1400
pUR

0.5257142857142857

In [124]:
pM_intUR =  404/1400
pM_intUR 

0.2885714285714286

De la ecuación de probabilidad condicional tenemos: 
$$p(UR|M)= \frac{p(UR\cap M)}{p(M)}=\frac{0.23}{0.49}=0.57$$   

In [125]:
pUR_dadoM = pM_intUR /pM 
pUR_dadoM 

0.5738636363636364

In [126]:
print(f"Es decir, la probabilidad de use redes para opinar de TV dado que es mujer es: {pUR_dadoM }")

Es decir, la probabilidad de use redes para opinar de TV dado que es mujer es: 0.5738636363636364


#### c) ¿Son independientes el género de la persona y el uso de las redes sociales y otros sitios web,con los fines mencionados en este problema? Justifique su respuesta

 Debido a que: 
 $$p(UR)\neq p(UR|M)$$


Se puede decir que NO son independientes el Uso de Redes Sociales (UR) con respecto al genero, por ejemplo, en este caso Mujeres

#### d) Su respuesta al inciso anterior ¿tiene alguna implicación acerca de la independencia o dependencia entre el género de la persona y el hecho de que no use las redes sociales ni otros sitios web, con los fines mencionados en este problema? Justifique su respuesta

Vamos a calcular como tal la probabilidad condicional de el NO uso de redes dado que es Mujer u Hombre: <br>
     $$p(NR|M)= \frac{p(NR\cap M)}{p(M)}$$  

In [127]:
pM_intNR =  300/1400
pM_intNR 

0.21428571428571427

In [128]:
pNR_dadoM = pM_intNR /pM 
pNR_dadoM 

0.4261363636363636

__Entonces se puede decir que__:_ Dado que es mujer, es más probable que SI use redes sociales para expresar su opinión respecto a peliculas a que no lo haga_

Ahora veamos que pasa si se analiza para el caso de los hombres:

La probabilidad de que No Use Redes para opinar de TV dado que es hombre se calcularía como: <br>
     $$p(NR|H)= \frac{p(NR\cap H)}{p(H)}$$ 

In [130]:
pH_intNR =  363/1400
pH_intNR

0.2592857142857143

In [131]:
pH = 696/1400
pH 

0.49714285714285716

In [132]:
pNR_dadoH = pH_intNR /pH 
pNR_dadoH 

0.521551724137931

La probabilidad de que SI Use Redes para opinar de TV dado que es hombre se calcularía como: <br>
     $$p(UR|H)= \frac{p(UR\cap H)}{p(H)}$$ 

In [133]:
pH_intUR =  332/1400
pH_intUR

0.23714285714285716

In [134]:
pUR_dadoH = pH_intUR /pH 
pUR_dadoH 

0.4770114942528736

__Entonces se puede decir que__:_Dado que es hombre, es más probable que NO use redes sociales para expresar su opinión respecto a peliculas a que si lo haga_

### Problema 4

Una compañía petrolera compró un terreno en Alaska. Realizó estudios geológicos preliminares y asignó las siguientes probabilidades: 50% para la probabilidad de encontrar petróleo de alta  calidad, el 20% de encontrar petróleo de calidad media, y el resto a la probabilidad de no encontrar petróleo.


a) Determine la probabilidad de encontrar petróleo.<br>
b) Se tomó una prueba de suelo después de 200 pies de perforación del primer pozo y las probabilidades de encontrar un tipo particular de suelo identificado con una prueba de seguimiento, son 0.20, si se encuentra petróleo de alta calidad, 0.80 si el petróleo  encontrado es de calidad media y 0.20 si no se encuentra petróleo. Cuando se encuentra el  tipo de suelo mencionado, ¿Cuáles son las probabilidades de obtener cada tipo de petróleo, o bien de no encontrar petróleo? Determine la probabilidad de encontrar petróleo en esta nueva circunstancia

__Definamos los eventos de nuestro espacio muestral S como:__<br>
    A: Probabilidad de encontrar petróleo de alta calidad<br>
    B: Probabilidad de encontrar petróleo de mala calidad<br>
    C: Probabilidad de NO encontrar petróleo<br>

Asignemos valores a nuestros eventos dentro de S

In [135]:
pA= 0.5
pA
pB= 0.2
pB
pC= 0.3
pC

0.3

#### a)  Determine la probabilidad de encontrar petróleo

Ahora definamos a D como:<br>
    D: Probabilidad de encontrar petróleo

In [136]:
pD = pA + pB
pD

0.7

In [137]:
print(f"Es decir, la probabilidad de SI encontrar petroleo es : {pD}")

Es decir, la probabilidad de SI encontrar petroleo es : 0.7


#### b)  Se tomó una prueba de suelo después de 200 pies de perforación del primer pozo y las probabilidades de encontrar un tipo particular de suelo identificado con una prueba de seguimiento, son 0.20, si se encuentra petróleo de alta calidad, 0.80 si el petróleo  encontrado es de calidad media y 0.20 si no se encuentra petróleo. Cuando se encuentra el  tipo de suelo mencionado, ¿Cuáles son las probabilidades de obtener cada tipo de petróleo, o bien de no encontrar petróleo? Determine la probabilidad de encontrar petróleo en esta nueva circunstancia

Ahora definamos a E como:<br>
    E:  Probabilidad de encontrar un tipo particular de suelo identificado con una prueba de seguimiento

Ahora se definene las probabilidades condicionales dadas en el enunciado 

$$p(E|A)= \frac{p(E\cap A)}{p(A)}=0.20$$ 

In [138]:
pE_dadoA = 0.2
pE_dadoA

0.2

$$p(E|B)= \frac{p(E\cap B)}{p(B)}=0.80$$ 

In [139]:
pE_dadoB = 0.8
pE_dadoB

0.8

$$p(E|C)= \frac{p(E\cap C)}{p(C)}=0.20$$ 

In [140]:
pE_dadoC = 0.2
pE_dadoC

0.2

Notar que la suma de p(E|A) p(E|B) + p(E|C) NO necesariamente debe de sumar 1, pues NO ESTPAN "EREFERNCIADAS" AL MISMO ESPACIO MUESTRAL S

Ahora, Dado que se encuentra el tipo de suelo, es decir dado E cual es probabilidad de obtener cada tipo de petróleo o NO encontrar petróleo 

__b.1) Prob de A dado E__ 

$$p(A|E)= \frac{p(A\cap E)}{p(E)}$$ 

Entonces para obtener la p(E) usemos de nuevo la "Formula de probabilidad Total"

 Sean A, B, C una partición colectivamente exhaustiva del espacio muestral S y sea E otro evento, entonces:

$$p(E)= p(E\cap A)\cup p(E\cap B)\cup p(E\cap C)$$ 

Calculemos la $p(E\cap A) = p(E|A)*p(A)=$

In [141]:
pE_intA = pE_dadoA * pA 
pE_intA 

0.1

Calculemos la $p(E\cap B) = p(E|B)*p(B)=$

In [142]:
pE_intB = pE_dadoB * pB 
pE_intB 

0.16000000000000003

Calculemos la $p(E\cap C) = p(E|C)*p(C)=$

In [143]:
pE_intC = pE_dadoC * pC
pE_intC 

0.06

Y entonces p(E) es:

In [144]:
pE = pE_intA + pE_intB + pE_intC 
pE

0.32

Por lo tanto, $$p(A|E)= \frac{p(A\cap E)}{p(E)}=$$ 

In [145]:
pA_dadoE = pE_intA /pE
pA_dadoE

0.3125

__b.2) Prob de B dado E__ 

 $$p(B|E)= \frac{p(B\cap E)}{p(E)}=$$ 

In [146]:
pB_dadoE = pE_intB /pE
pB_dadoE

0.5000000000000001

__b.3) Prob de C dado E__ 

 $$p(C|E)= \frac{p(C\cap E)}{p(E)}=$$ 

In [147]:
pC_dadoE = pE_intC /pE
pC_dadoE

0.1875

La probabilidad de encontrar petroleo en estas circusntancias es: 

Ahora definamos a F como:<br>
    F:  Probabilidad de encontrar petroleo dado que E pasó

In [148]:
pF = pA_dadoE +pB_dadoE
pF

0.8125000000000001

In [149]:
print(f"Es decir, la probabilidad de SI encontrar petroleo en estas nuevas circusntancias es : {pF}")

Es decir, la probabilidad de SI encontrar petroleo en estas nuevas circusntancias es : 0.8125000000000001


Que es mayor que la pD :Probabilidad de encontrar petróleo en las "circusntancias originales"

### Problema 5

Resuelva el siguiente caso: .<br>
JUECES DEL CONDADO DE HAMILTON.<br>
.<br>
Los jueces del condado de Hamilton procesan miles de casos al año. En la gran mayoría de los  casos desechados, el veredicto permanece como se presentó. Sin embargo, algunos son 
apelados y de éstos algunos se revocan. Kristen DelGuzzi, del diario Cincinnati Enquirer, realizó  un estudio de los casos manejados por los jueces del condado de Hamilton durante un periodo 
de tres años (Cincinnati Enquirer, 11 de enero de 1998). En la tabla que se muestra a  continuación se muestran los resultados de 182,908 casos manejados por 38 jueces del tribunal 
de primera instancia (Common Pleas Court), del tribunal de lo familiar (Domestic Relations  Court) y del tribunal municipal (Municipal Court). Dos de los jueces (Dinkelacker y Hogan) no 
trabajaron en el mismo tribunal durante los tres años, sino que en algún momento de eses  período cambiaron de tribunal.<br>
<br>
El propósito del estudio del periódico es evaluar el desempeño de los jueces. Las apelaciones con frecuencia son el resultado de los errores cometidos por éstos, y el periódico quería saber cuáles 
de ellos hacían un buen trabajo y cuáles cometían demasiados errores. A usted le llaman para  que ayude en el análisis de datos. Utilice sus conocimientos de probabilidad y probabilidad 
condicional para ayudar a calificar a los jueces. Tal vez pueda analizar la probabilidad de los casos manejados en los diferentes tribunales que fueron apelados y revocados.<br>
<br>
Elabore un informe con sus calificaciones de los jueces. Incluya también un análisis de la probabilidad de apelación y la revocación de casos en los tres tribunales. Como mínimo, su  informe debe incluir lo siguiente:


• La probabilidad de casos apelados (Appealed Cases) y revocados (Reversed Cases) en los 
tres tribunales.<br>
• La probabilidad de que un caso sea apelado, por cada juez.<br>
• La probabilidad de que un caso sea revocado, por cada juez.<br>
• La probabilidad de una revocación, dada una apelación, por cada juez.<br>
• Una clasificación de los jueces dentro de cada tribunal. Establezca los criterios que manejó 
y las razones de su elección

Primero vamos a importar el excel que contiene la información de los jueces y sus casos

In [150]:
import pandas as pd

# Read the Excel file into a dictionary of data frames
dfs = pd.read_excel(R'C:\Users\sergi\OneDrive\Documentos\MIS_UNAM\Segundo_semestre\Estocasticos_UNAM\Estocasticos_Tarea1_Fundamentos_Probabilidad_datos_problema5.xlsx', sheet_name=['Common_please', 'Domestic_relations', 'Municipal'])
print(dfs)
type(dfs)


{'Common_please':                     Judge  Total cases Disposed  Appealed Cases  \
0          Fred Cartolano                  3037             137   
1            Thomas Crush                  3372             119   
2     Patrick Dinkelacker                  1258              44   
3           Timothy Hogan                  1954              60   
4            Robert Kraft                  3138             127   
5         William Mathews                  2264              91   
6       William Morrissey                  3032             121   
7           Norbert Nadel                  2959             131   
8         Arthur Ney, Jr.                  3219             125   
9         Richard Niehaus                  3353             137   
10           Thomas Nurre                  3000             121   
11          John O'Connor                  2969             129   
12        Robert Ruchlman                  3205             145   
13  J. Howard Sunderrnann                   

dict

### Calculemos las probabilidades para el caso de los jueces de Common Please Court

In [151]:
# Access each data frame using the sheet name as the key
df_common = dfs['Common_please']
print(df_common)
type(df_common)


                    Judge  Total cases Disposed  Appealed Cases  \
0          Fred Cartolano                  3037             137   
1            Thomas Crush                  3372             119   
2     Patrick Dinkelacker                  1258              44   
3           Timothy Hogan                  1954              60   
4            Robert Kraft                  3138             127   
5         William Mathews                  2264              91   
6       William Morrissey                  3032             121   
7           Norbert Nadel                  2959             131   
8         Arthur Ney, Jr.                  3219             125   
9         Richard Niehaus                  3353             137   
10           Thomas Nurre                  3000             121   
11          John O'Connor                  2969             129   
12        Robert Ruchlman                  3205             145   
13  J. Howard Sunderrnann                   955              6

pandas.core.frame.DataFrame

Calculemos la probabilidad de que un caso se apelado por cada juez de Common Please Court 

In [152]:
df_common['Probabilidad_de_apleacion'] = df_common['Appealed Cases'] / df_common['Total cases Disposed'] 
print(df_common['Probabilidad_de_apleacion'])

0     0.045110
1     0.035291
2     0.034976
3     0.030706
4     0.040472
5     0.040194
6     0.039908
7     0.044272
8     0.038832
9     0.040859
10    0.040333
11    0.043449
12    0.045242
13    0.062827
14    0.040433
15    0.028488
16    0.040096
Name: Probabilidad_de_apleacion, dtype: float64


In [153]:
# Rename a column in the df_common data frame
df_common = df_common.rename(columns={'Probabilidad_de_apleacion': 'Probabilidad_de_apelacion'})

In [154]:
print(df_common)

                    Judge  Total cases Disposed  Appealed Cases  \
0          Fred Cartolano                  3037             137   
1            Thomas Crush                  3372             119   
2     Patrick Dinkelacker                  1258              44   
3           Timothy Hogan                  1954              60   
4            Robert Kraft                  3138             127   
5         William Mathews                  2264              91   
6       William Morrissey                  3032             121   
7           Norbert Nadel                  2959             131   
8         Arthur Ney, Jr.                  3219             125   
9         Richard Niehaus                  3353             137   
10           Thomas Nurre                  3000             121   
11          John O'Connor                  2969             129   
12        Robert Ruchlman                  3205             145   
13  J. Howard Sunderrnann                   955              6

Calculemos la probabilidad de que un caso se revocado por cada juez de Common Please Court

In [156]:
df_common['Probabilidad_de_revocacion'] = df_common['Reversed cases'] / df_common['Total cases Disposed'] 
print(df_common['Probabilidad_de_revocacion'])

0     0.003951
1     0.002966
2     0.006359
3     0.003582
4     0.002231
5     0.007951
6     0.007256
7     0.006759
8     0.004349
9     0.004772
10    0.002000
11    0.004042
12    0.005616
13    0.010471
14    0.004139
15    0.001942
16    0.004528
Name: Probabilidad_de_revocacion, dtype: float64


__SE PRESENTA A CONTINUACION LA TABLA RESUMEN DE LAS PROBABILIDADES INDIVIDUALES Y PROMEDIO DE APLEACIÓN Y REVOCACIÓN PARA CADA JUEZ DE LA CORTE DE COMMON PLEASE__

In [157]:
print(df_common)

                    Judge  Total cases Disposed  Appealed Cases  \
0          Fred Cartolano                  3037             137   
1            Thomas Crush                  3372             119   
2     Patrick Dinkelacker                  1258              44   
3           Timothy Hogan                  1954              60   
4            Robert Kraft                  3138             127   
5         William Mathews                  2264              91   
6       William Morrissey                  3032             121   
7           Norbert Nadel                  2959             131   
8         Arthur Ney, Jr.                  3219             125   
9         Richard Niehaus                  3353             137   
10           Thomas Nurre                  3000             121   
11          John O'Connor                  2969             129   
12        Robert Ruchlman                  3205             145   
13  J. Howard Sunderrnann                   955              6

In [158]:

probabilidad_Casos_apelados_corte_common = (df_common.at[16, 'Probabilidad_de_apelacion'])*100
print(f" La probabilidad 'promedio' de casos apelados en la corte Common Please es: {probabilidad_Casos_apelados_corte_common} %")


 La probabilidad 'promedio' de casos apelados en la corte Common Please es: 4.009557401297076 %


In [159]:
probabilidad_Casos_revocados_corte_common = df_common.at[16, 'Probabilidad_de_revocacion']*100
print(f" La probabilidad 'promedio' de casos revocados en la corte Common Please es: {probabilidad_Casos_revocados_corte_common} %")

 La probabilidad 'promedio' de casos revocados en la corte Common Please es: 0.45283877574240533 %


__SE MUESTRAN LOS JUECES DE LA CORTE COMMON PLEASE, DE AQUEL CON MAYOR APELACIÓN A AQUEL CON MENOR APELACIÓN__

In [160]:
# Sort the df_common data frame by the Probabilidad_de_apelacion column in descending order
df_common_apelacion_decendiente = df_common.sort_values(by='Probabilidad_de_apelacion', ascending=False)
print(df_common_apelacion_decendiente)

                    Judge  Total cases Disposed  Appealed Cases  \
13  J. Howard Sunderrnann                   955              60   
12        Robert Ruchlman                  3205             145   
0          Fred Cartolano                  3037             137   
7           Norbert Nadel                  2959             131   
11          John O'Connor                  2969             129   
9         Richard Niehaus                  3353             137   
4            Robert Kraft                  3138             127   
14       Ann Marie Tracey                  3141             127   
10           Thomas Nurre                  3000             121   
5         William Mathews                  2264              91   
16                  Total                 43945            1762   
6       William Morrissey                  3032             121   
8         Arthur Ney, Jr.                  3219             125   
1            Thomas Crush                  3372             11

### Calculemos las probabilidades para el caso de los jueces de Domestic Relations Court

In [161]:
df_domestic = dfs['Domestic_relations']
print(df_domestic)
type(df_domestic)

                 Judge  Total Cases Disposed  Appealed Cases  Reversed Cases
0  Penelope Cunningham                  2729               7               1
1  Patrick Dinkelacker                  6001              19               4
2       Deborah Gaines                  8799              48               9
3       Ronald Panioto                 12970              32               3
4                Total                 30499             106              17


pandas.core.frame.DataFrame

Calculemos la probabilidad de que un caso se apelado por cada juez de Domestic Relations Court 

In [166]:
df_domestic['Probabilidad_de_apleacion'] = df_domestic['Appealed Cases'] / df_domestic['Total Cases Disposed'] 
print(df_domestic['Probabilidad_de_apleacion'])

0    0.002565
1    0.003166
2    0.005455
3    0.002467
4    0.003476
Name: Probabilidad_de_apleacion, dtype: float64


Calculemos la probabilidad de que un caso se revocado por cada juez de Domestic Relations Court 

In [167]:
df_domestic['Probabilidad_de_revocacion'] = df_domestic['Reversed Cases'] / df_domestic['Total Cases Disposed'] 
print(df_domestic['Probabilidad_de_revocacion'])

0    0.000366
1    0.000667
2    0.001023
3    0.000231
4    0.000557
Name: Probabilidad_de_revocacion, dtype: float64


__SE PRESENTA A CONTINUACION LA TABLA RESUMEN DE LAS PROBABILIDADES INDIVIDUALES Y PROMEDIO DE APLEACIÓN Y REVOCACIÓN PARA CADA JUEZ DE LA CORTE DE DOMESTIC RELATIONS__

In [168]:
print(df_domestic)

                 Judge  Total Cases Disposed  Appealed Cases  Reversed Cases  \
0  Penelope Cunningham                  2729               7               1   
1  Patrick Dinkelacker                  6001              19               4   
2       Deborah Gaines                  8799              48               9   
3       Ronald Panioto                 12970              32               3   
4                Total                 30499             106              17   

   Probabilidad_de_apleacion  Probabilidad_de_revocacion  
0                   0.002565                    0.000366  
1                   0.003166                    0.000667  
2                   0.005455                    0.001023  
3                   0.002467                    0.000231  
4                   0.003476                    0.000557  


In [169]:
probabilidad_Casos_apelados_corte_domestic = (df_domestic.at[4, 'Probabilidad_de_apleacion'])*100
print(f" La probabilidad 'promedio' de casos apelados en la corte Domestic es: {probabilidad_Casos_apelados_corte_domestic} %")


 La probabilidad 'promedio' de casos apelados en la corte Domestic es: 0.34755237876651696 %


In [170]:
probabilidad_Casos_revocados_corte_domestic = df_domestic.at[4, 'Probabilidad_de_revocacion']*100
print(f" La probabilidad 'promedio' de casos revocados en la corte Domestic es: {probabilidad_Casos_revocados_corte_domestic} %")

 La probabilidad 'promedio' de casos revocados en la corte Domestic es: 0.055739532443686686 %


__SE MUESTRAN LOS JUECES DE LA CORTE DOMESTIC, DE AQUEL CON MAYOR APELACIÓN A AQUEL CON MENOR APELACIÓN__

In [171]:
# Sort the df_common data frame by the Probabilidad_de_apelacion column in descending order
df_domestic_apelacion_decendiente = df_domestic.sort_values(by='Probabilidad_de_apleacion', ascending=False)
print(df_domestic_apelacion_decendiente)

                 Judge  Total Cases Disposed  Appealed Cases  Reversed Cases  \
2       Deborah Gaines                  8799              48               9   
4                Total                 30499             106              17   
1  Patrick Dinkelacker                  6001              19               4   
0  Penelope Cunningham                  2729               7               1   
3       Ronald Panioto                 12970              32               3   

   Probabilidad_de_apleacion  Probabilidad_de_revocacion  
2                   0.005455                    0.001023  
4                   0.003476                    0.000557  
1                   0.003166                    0.000667  
0                   0.002565                    0.000366  
3                   0.002467                    0.000231  


### Calculemos las probabilidades para el caso de los jueves de Municipal Court

In [172]:
df_municipal = dfs['Municipal']
print(df_municipal)
type(df_municipal)

                   Judge  Total Cases Disposed  Appealed Cases  Reversed Cases
0             Mike Allen                  6149              43               4
1           Nadine Allen                  7812              34               6
2          Timothy Black                  7954              41               6
3            David Davis                  7736              43               5
4   Leslie Isaiah Gaines                  5282              35              13
5            Karla Grady                  5253               6               0
6            Deidra Hair                  2532               5               0
7         Dennis Helmick                  7900              29               5
8          Timothy Hogan                  2308              13               2
9    James PatrCk Kenney                  2798               6               1
10       Joseph Luebbers                  4698              25               8
11       William Mallory                  8277      

pandas.core.frame.DataFrame

Calculemos la probabilidad de que un caso se apelado por cada juez de Municipal Court 

In [174]:
df_municipal['Probabilidad_de_apleacion'] = df_municipal['Appealed Cases'] / df_municipal['Total Cases Disposed'] 
print(df_municipal['Probabilidad_de_apleacion'])

0     0.006993
1     0.004352
2     0.005155
3     0.005558
4     0.006626
5     0.001142
6     0.001975
7     0.003671
8     0.005633
9     0.002144
10    0.005321
11    0.004591
12    0.004137
13    0.004376
14    0.005628
15    0.003126
16    0.005263
17    0.006108
18    0.004096
19    0.001430
20    0.004610
Name: Probabilidad_de_apleacion, dtype: float64


Calculemos la probabilidad de que un caso se revocado por cada juez de Muncipal Court 

In [177]:
df_municipal['Probabilidad_de_revocacion'] = df_municipal['Reversed Cases'] / df_municipal['Total Cases Disposed'] 
print(df_municipal['Probabilidad_de_revocacion'])

0     0.000651
1     0.000768
2     0.000754
3     0.000646
4     0.002461
5     0.000000
6     0.000000
7     0.000633
8     0.000867
9     0.000357
10    0.001703
11    0.001087
12    0.000852
13    0.000337
14    0.001809
15    0.001340
16    0.001669
17    0.001110
18    0.000745
19    0.000715
20    0.000959
Name: Probabilidad_de_revocacion, dtype: float64


__SE PRESENTA A CONTINUACION LA TABLA RESUMEN DE LAS PROBABILIDADES INDIVIDUALES DE APLEACIÓN Y REVOCACIÓN PARA CADA JUEZ DE LA CORTE DE MUNICIPAL__

In [178]:
print(df_municipal)

                   Judge  Total Cases Disposed  Appealed Cases  \
0             Mike Allen                  6149              43   
1           Nadine Allen                  7812              34   
2          Timothy Black                  7954              41   
3            David Davis                  7736              43   
4   Leslie Isaiah Gaines                  5282              35   
5            Karla Grady                  5253               6   
6            Deidra Hair                  2532               5   
7         Dennis Helmick                  7900              29   
8          Timothy Hogan                  2308              13   
9    James PatrCk Kenney                  2798               6   
10       Joseph Luebbers                  4698              25   
11       William Mallory                  8277              38   
12           Melba Marsh                  8219              34   
13        Beth Mattingly                  2971              13   
14     Alb

In [180]:
probabilidad_Casos_apelados_corte_municipal = (df_municipal.at[20, 'Probabilidad_de_apleacion'])*100
print(f" La probabilidad 'promedio' de casos apelados en la corte municipal es: {probabilidad_Casos_apelados_corte_municipal} %")

 La probabilidad 'promedio' de casos apelados en la corte municipal es: 0.46098244578846437 %


In [181]:
probabilidad_Casos_revocados_corte_municipal = (df_municipal.at[20, 'Probabilidad_de_revocacion'])*100
print(f" La probabilidad 'promedio' de casos revocados en la corte municipal es: {probabilidad_Casos_revocados_corte_municipal} %")

 La probabilidad 'promedio' de casos revocados en la corte municipal es: 0.09588434872400059 %


__SE MUESTRAN LOS JUECES DE LA CORTE MUNICIPAL, DE AQUEL CON MAYOR APELACIÓN A AQUEL CON MENOR APELACIÓN__

In [182]:
# Sort the df_common data frame by the Probabilidad_de_apelacion column in descending order
df_municipal_apelacion_decendiente = df_municipal.sort_values(by='Probabilidad_de_apleacion', ascending=False)
print(df_municipal_apelacion_decendiente)

                   Judge  Total Cases Disposed  Appealed Cases  \
0             Mike Allen                  6149              43   
4   Leslie Isaiah Gaines                  5282              35   
17       Mark Schweikert                  5403              33   
8          Timothy Hogan                  2308              13   
14     Albert Mestemaker                  4975              28   
3            David Davis                  7736              43   
10       Joseph Luebbers                  4698              25   
16            Jack Rosen                  7790              41   
2          Timothy Black                  7954              41   
20                 Total                108464             500   
11       William Mallory                  8277              38   
13        Beth Mattingly                  2971              13   
1           Nadine Allen                  7812              34   
12           Melba Marsh                  8219              34   
18        